src/Client.php

```diff
@@ -201,4 +192,24 @@ private function prepareAttributes(array $data)
 
         return $data;
     }
+
+    /**
+     * Validate endPoint is a valid URL
+     *
+     * @param [array] $options
+     * @param [Validator] $validator
+     * @return void
+     */
+    private static function validateClientEndpoint($options, $validator){
+        $violations = $validator->validate($options['apiEndpoint'], [
+            new Url(),
+        ]);
+        if(count($violations) != 0){
+            throw new Exception\InvalidAttributeValue(
+                'Your API endpoint is not a valid URL',
+                $violations
+            );
+        }
+        Client::$url = $options['apiEndpoint'];
```

Comment:
Assigning the new parameter should not be a responsibility of a method that does validation. I would move this to line 68, after the validator returns without errors